### PR TITLE
Fix for char and string literals as function args

### LINKED ISSUE #671: Compiler error for inner function call with char argument
**Minimal example**: Compile the following code with Clang++ with the clad plugin:

```c++
#include "clad/Differentiator/Differentiator.h"

void g(char c){}

double f(double x) {
  g('a');
  return x;
}

int main() {
  auto f_dx = clad::differentiate(f, "x");
  f_dx.execute(3);
}
```

**Expected behavior**: Code compiles.

**Observed behavior**: Compiler error:

```
charcall.cpp:6:5: warning: attempted to differentiate unsupported statement, no changes applied
  g('a');
    ^
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
...
clang: error: unable to execute command: Segmentation fault (core dumped)
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Ubuntu clang version 11.1.0-6
...
```

**Further information**: The error goes away when the char literal is replaced by an integer literal. The problem also shows up when a string literal is provided for a `char*` argument (which is the setup where I stumbled across this bug, but the above case shows that it's not related to pointers). Many thanks to @vgvassilev and @parth-07 for your quick responses to my earlier bug reports. I'm trying to apply Clad to a particle physics simulation code with as few changes to that code as possible, which seems to break a few things...

**Setup**: Clad built from the GitHub master branch at commit 1bdd261 on Ubuntu 22.04 with Clang 11.

### PR DESCRIPTION
**Pull request by vaithak**
fixes #671

### CODE REVIEW

File: lib/Differentiator/DiffPlanner.cpp [hunk 1]
```diff
@@ -562,9 +562,14 @@ namespace clad {
...
-             FD->getTemplateSpecializationArgs()->get(0).pack_elements()) {
-          bitmasked_opts_value |= arg.getAsIntegral().getExtValue();
+        const auto template_arg = FD->getTemplateSpecializationArgs()->get(0);
+        if (template_arg.getKind() == TemplateArgument::Pack) {
+          for (const auto& arg :
+               FD->getTemplateSpecializationArgs()->get(0).pack_elements()) {
+            bitmasked_opts_value |= arg.getAsIntegral().getExtValue();
+          }
+        } else {
+          bitmasked_opts_value = template_arg.getAsIntegral().getExtValue();
```
Comment (vaithak) [inline lib/Differentiator/DiffPlanner.cpp:572]: codecov reports this as missing coverage, as it was only required for the `build_release` version.

File: lib/Differentiator/DiffPlanner.cpp [hunk 2]
```diff
@@ -602,9 +607,14 @@ namespace clad {
...
```
Comment (vaithak) [inline lib/Differentiator/DiffPlanner.cpp:617]: same here.

File: lib/Differentiator/DiffPlanner.cpp [hunk 3]
```diff
@@ -562,9 +562,14 @@ namespace clad {
...
         unsigned bitmasked_opts_value = 0;
-        for (auto const& arg :
-             FD->getTemplateSpecializationArgs()->get(0).pack_elements()) {
-          bitmasked_opts_value |= arg.getAsIntegral().getExtValue();
+        const auto template_arg = FD->getTemplateSpecializationArgs()->get(0);
+        if (template_arg.getKind() == TemplateArgument::Pack) {
+          for (const auto& arg :
+               FD->getTemplateSpecializationArgs()->get(0).pack_elements()) {
+            bitmasked_opts_value |= arg.getAsIntegral().getExtValue();
+          }
```
Comment (vgvassilev) [inline lib/Differentiator/DiffPlanner.cpp:570]:
```suggestion
          for (const auto& arg :
               FD->getTemplateSpecializationArgs()->get(0).pack_elements())
            bitmasked_opts_value |= arg.getAsIntegral().getExtValue();
```

File: lib/Differentiator/DiffPlanner.cpp [hunk 4]
```diff
@@ -562,9 +562,14 @@ namespace clad {
...
+        } else {
```
Comment (vgvassilev) [inline lib/Differentiator/DiffPlanner.cpp:571]: We can drop the braces here too...

**Comment by vaithak** [comment]
The coverage failure is because of statements which executes for debug and build_release versions as mentioned above.

**Review by vgvassilev — APPROVED** [review]
Codecov seems confusing itself. LGTM!

**Pull request merged**